### DISCUSSION TITLE
Future Plans for a Caching Subsystem?

### DISCUSSION THREAD

**Discussion by Kurozeto** [jellyfin/jellyfin-android#472] [General]
The Jellyfin project has made leaps and strides in the recent year. As more people continue to take on the project of home media servers, and access them outside of their home, it is with interest that various applications implement subtle improvements to their back-end. 

In order to help decrease load times and mobile data usage, I would like to propose the implementation of a user-selective caching subsystem on a per-library basis. The benefits of the sub-system will greatly increase responsiveness of the application, especially in instances relating to music playback. The primary use-case would be in the instance of track skipping, music queuing, and audio on repeat.

While this will mainly benefit those who use Jellyfin on the go for audio media consumption, I'm sure there are other use cases within the application that this subsystem could be utilized to increase performance. If possible, I would like to continue this discussion, and even put together a proposal for implementation if it gains traction.

**Comment by nielsvanvelzen** [comment]
A feature request for offline mode is here: https://features.jellyfin.org/posts/218/support-offline-mode-on-android-mobile

Right now it's not possible to implement in jf-android since it's mostly just a webview, when the app uses a native UI etc. we could look into storing media offline in some way, but it's still a lot of work.

**Comment by Morgan-6Freedom** [comment]
Is there a way to download/cache the next music in the playlist while the current music is playing (with the current codebase) ? It would be great when you are in your car and the network is down for a couple of minutes.
